src/attribution/onINP.ts

```diff
@@ -233,6 +235,100 @@ const getIntersectingLoAFs = (
   return intersectingLoAFs;
 };
 
+const getLoAFSummary = (attribution: INPAttribution) => {
+  const loafAttribution: LongAnimationFrameSummary = {};
+
+  // Stats across all LoAF entries and scripts.
+  const interactionTime = attribution.interactionTime;
+  const inputDelay = attribution.inputDelay;
+  const processingDuration = attribution.processingDuration;
+  let totalNonForcedStyleAndLayoutDuration = 0;
+  let totalForcedStyleAndLayout = 0;
+  let totalScriptTime = 0;
+  let numScripts = 0;
+  let slowestScriptDuration = 0;
+  let slowestScriptEntry!: PerformanceScriptTiming;
+  let slowestScriptPhase: string = '';
+  const phases = {} as Record<string, Record<ScriptInvokerType, number>>;
+
+  attribution.longAnimationFrameEntries.forEach((loafEntry) => {
+    totalNonForcedStyleAndLayoutDuration +=
+      loafEntry.startTime + loafEntry.duration - loafEntry.styleAndLayoutStart;
+    loafEntry.scripts.forEach((script) => {
+      const scriptEndTime = script.startTime + script.duration;
+      if (scriptEndTime < interactionTime) {
+        return;
+      }
+      const intersectingScriptDuration =
+        scriptEndTime - Math.max(interactionTime, script.startTime);
+      totalScriptTime += intersectingScriptDuration;
+      numScripts++;
+      totalForcedStyleAndLayout += script.forcedStyleAndLayoutDuration;
+      const invokerType = script.invokerType;
+      let phase = 'processingDuration';
+      if (script.startTime < interactionTime + inputDelay) {
+        phase = 'inputDelay';
+      } else if (
+        script.startTime >
+        interactionTime + inputDelay + processingDuration
+      ) {
+        phase = 'presentationDelay';
+      }
+
+      // Define the record if necessary
+      phases[phase] ??= {} as Record<ScriptInvokerType, number>;
+      phases[phase][invokerType] ??= 0;
+      // Increment it with this value
+      phases[phase][invokerType] += intersectingScriptDuration;
+
+      if (intersectingScriptDuration > slowestScriptDuration) {
+        slowestScriptEntry = script;
+        slowestScriptPhase = phase;
+        slowestScriptDuration = intersectingScriptDuration;
+      }
+    });
+  });
+
+  // Gather the summary information into the loafAttribution object
+  loafAttribution.numLongAnimationFrames =
+    attribution.longAnimationFrameEntries.length;
+  loafAttribution.numIntersectingScripts = numScripts;
+  if (slowestScriptEntry !== null) {
+    const slowestScript: SlowestScriptSummary = {
+      entry: slowestScriptEntry,
+      phase: slowestScriptPhase as
+        | 'inputDelay'
+        | 'processingDuration'
+        | 'presentationDelay',
```

Comment:
this is actually

```suggestion
      phase: slowestScriptPhase as
        | 'inputDelay'
        | 'processingDuration'
        | 'presentationDelay'
        | '',
```
?

Reply:
Removed this typing. It should always be defined if a script exists.